shapely/linref.py

```diff
@@ -1,14 +1,15 @@
 """Linear referencing
 """
 
+from shapely.errors import GeometryTypeError
 from shapely.topology import Delegating
 
 
 class LinearRefBase(Delegating):
     def _validate_line(self, ob):
         super(LinearRefBase, self)._validate(ob)
         if not ob.geom_type in ['LinearRing', 'LineString', 'MultiLineString']:
-            raise TypeError("Only linear types support this operation")
+            raise GeometryTypeError("Only linear types support this operation")
```

Comment:
Yes.

Reply:
In PyGEOS, this actually returns a GEOSException, xref the discussion in https://github.com/Toblerity/Shapely/issues/991